### PR TITLE
Recruitment Task + Auxiliary Files

### PR DESCRIPTION
**Pull request by dhunanyan**
Hope for the best

### CODE REVIEW

File: src/static/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+declare module '*.svg' {
```
Comment (awojciak) [inline src/static/index.d.ts:1]: Had you saw some errors with types of images imports? I wonder why did you add type definitions for them.

File: src/static/icons.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ import ANAIcon from './icons/ana.svg'
 import activeIcon from './svg/active.svg'
 import inactiveIcon from './svg/inactive.svg'
 import allIcon from './svg/all.svg'
+import { LocalPhoneTwoTone } from '@material-ui/icons'
```
Comment (awojciak) [inline src/static/icons.ts:38]: unused import

File: src/TestTask/style.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    flexWrap: 'nowrap',
+    maxWidth: 740,
+    margin: '96px auto 126px auto',
+    minHeight: '100%',
+    maxHeight: 430,
+
+    ['@media(max-width: 580px)']: {
+      flexDirection: 'column',
+      maxHeight: '100%',
+      margin: '0px auto 0px auto'
```
Comment (awojciak) [inline src/TestTask/style.ts:36]: '0 auto' would be enough

File: src/TestTask/style.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { makeStyles } from '@material-ui/core/styles'
+import { colors, typography } from '@static/theme'
+import { findLastIndex } from 'lodash'
+
+const useStyles = makeStyles(() => ({
+  rootBackground: {
+    background: '#0b090d',
```
Comment (awojciak) [inline src/TestTask/style.ts:7]: use colors from theme when possible

File: src/TestTask/components/Wallet/Wallet.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            currency={currencyType}
+            value={value}
+            currencyRate={toCurrency[currentCurrency].rate}
+            toCurrency={toCurrency[currentCurrency].name}
+            currencyIconSrc={currencyType === 'SNY' ? undefined : icons['SNY']}
+          />
+        </Grid>
+
+        <AnimatedButton
+          onClick={() => {}}
```
Comment (awojciak) [inline src/TestTask/components/Wallet/Wallet.tsx:88]: this should be set through props

File: src/TestTask/components/Stats/Stats.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import { useAddonState } from '@storybook/store'
+
+import Stats from './Stats'
+
+storiesOf('test/components/Stats', module)
+  .addDecorator(withKnobs)
+  .add('default', () => {
+    const [data, setData] = useAddonState<Array<{ timestamp: number; value: number }>>('time', [])
+
+    React.useEffect(() => {
```
Comment (awojciak) [inline src/TestTask/components/Stats/Stats.stories.tsx:13]: I think you should use storybook hooks inside stories

File: src/TestTask/components/Header/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,179 @@
+import React from 'react'
```
Comment (awojciak) [inline src/TestTask/components/Header/Header.tsx:1]: I wonder why did you create a whole new header here? It wasn't part of the task.

Comment (awojciak) [reply src/TestTask/components/Header/Header.tsx:1]: Oh, I see now, you've created whole new directory for this task. It was unncecessary, subdirectory in 'components' for component from mockups would be enough

Comment (awojciak) [reply src/TestTask/components/Header/Header.tsx:1]: there is too much redundant, copied code

File: src/TestTask/components/Stats/Stats.stories.tsx (hunk 6 above)
Comment (awojciak) [reply src/TestTask/components/Stats/Stats.stories.tsx:13]: also, you're setting data only once, hooks usage seems to be unnecessary

File: src/TestTask/App.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import { useAddonState } from '@storybook/store'
+
+import { BN } from '@project-serum/anchor'
+import { printBN } from '@consts/utils'
+
+storiesOf('test/App', module)
+  .addDecorator(story => <MemoryRouter initialEntries={['/']}>{story()}</MemoryRouter>)
+  .add('default', () => {
+    const balance = printBN(new BN(100).mul(new BN(4603445)), 6)
+    const [data, setData] = useAddonState<Array<{ timestamp: number; value: number }>>('time', [])
```
Comment (awojciak) [inline src/TestTask/App.stories.tsx:19]: next case of unnecessary usage of hooks

File: src/TestTask/components/Amount/Amount.tsx [hunk 9]
```diff
@@ -0,0 +1,253 @@
+import React, { useRef, CSSProperties } from 'react'
```
Comment (awojciak) [inline src/TestTask/components/Amount/Amount.tsx:1]: next component, that already exists

File: src/TestTask/components/AnimatedButton/AnimatedButton.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
+import React from 'react'
```
Comment (awojciak) [inline src/TestTask/components/AnimatedButton/AnimatedButton.tsx:1]: end next one copied

File: src/TestTask/components/Plot/Plot.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { withKnobs } from '@storybook/addon-knobs'
+import { storiesOf } from '@storybook/react'
+import { useAddonState } from '@storybook/store'
+import Plot from './Plot'
+
+storiesOf('test/components/Plot', module)
+  .addDecorator(withKnobs)
+  .add('liquidity', () => {
+    const [data, setData] = useAddonState<Array<{ timestamp: number; value: number }>>('time', [])
```
Comment (awojciak) [inline src/TestTask/components/Plot/Plot.stories.tsx:10]: next case of unnecessary usage of hooks

File: src/TestTask/components/CalculatorSwitch/CalculatorSwitch.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+  setCurrentCurrency
+}) => {
+  const isXs = useMediaQuery(theme.breakpoints.down('xs'))
+
+  const tabsClasses = useTabsStyles({ value: currentCurrency })
+  const singleTabClasses = useSingleTabStyles()
+
+  const handleChange = (_: React.ChangeEvent<{}>, newValue: number) => {
+    setCurrentCurrency(newValue)
+    onSwitch(!newValue)
```
Comment (awojciak) [inline src/TestTask/components/CalculatorSwitch/CalculatorSwitch.tsx:28]: there are three values in this switch, so this line will not work properly here; I know that you've copied it from other component, but that was other case, where there are only two values

File: src/TestTask/components/CalculatorValue/CalculatorValue.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+}) => {
+  const classes = useStyles()
+
+  return (
+    <Grid container className={classes.root}>
+      <Grid container className={classes.container}>
+        {/* INPUT VALUE */}
+        <Typography variant='h3' className={`${classes.title} ${!value && classes.titleEmpty}`}>{`${
+          value ? value : '0.0'
+        } ${currency === 'AERGO' ? 'ARG' : currency}`}</Typography>
```
Comment (awojciak) [inline src/TestTask/components/CalculatorValue/CalculatorValue.tsx:32]: could you explain why are you using this 'AERGO'? I don't see anything like that on mockups

File: src/TestTask/components/Info/style.ts [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+    '&:last-of-type': {
+      margin: '4px 0 0 0'
+    }
+  },
+
+  title: {
+    margin: '0 4px 0 0',
+    color: '#a9b6bf',
+    fontFamily: 'Mukta',
```
Comment (awojciak) [inline src/TestTask/components/Info/style.ts:28]: setting fontFamilly should be unnecessary if only you're using Typography component instead of normal divs/spans

File: src/TestTask/components/Info/style.ts [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    fontStyle: 'normal',
+    fontWeight: 400,
```
Comment (awojciak) [inline src/TestTask/components/Info/style.ts:30]: use typography from config when possible

File: src/TestTask/components/Plot/Plot.tsx [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+  let minY = 0
+  let maxY = 0
+  const dateAndHours = data.map(({ timestamp, value }) => {
+    if (rate) {
+      minY =
+        rate > 0
+          ? value * rate < minY
+            ? value * rate
+            : minY
+          : value * rate < minY
```
Comment (awojciak) [inline src/TestTask/components/Plot/Plot.tsx:40]: it's repeated previous case

File: src/TestTask/components/Wallet/Wallet.tsx [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+            placeholder='0.0'
+            changeCurrencyType={changeCurrencyType}
+          />
+        </Grid>
+        <Grid container className={classes.calculatorValueSwitch}>
+          <Typography variant='h5' className={classes.subSubtitle}>
+            Calculator
+          </Typography>
+          <CalculatorSwitch
+            onSwitch={() => {}}
```
Comment (awojciak) [inline src/TestTask/components/Wallet/Wallet.tsx:72]: it should do something

**Review by awojciak — CHANGES_REQUESTED** [review]
One more issue: max button in input doesn't work.
Also there are some deviations from mockups:
obraz
obraz

Otherwise component looks good. Remember to not copy existing components, when it's unnecessary. When you're copying existing code to use it somewhere else, don't leave it in unchanged form; do some modifications to make it work properly in that specific case

File: src/static/index.d.ts (hunk 1 above)
Comment (dhunanyan) [reply src/static/index.d.ts:1]: Indeed, I was getting errors when importing images. Error were also noticed for earlier imported images, too, so I decided to add type definitions to fix it and avoid unnecessary warnings.

File: src/static/icons.ts (hunk 2 above)
Comment (dhunanyan) [reply src/static/icons.ts:38]: Agree, missed to remove.

File: src/TestTask/style.ts [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'space-between',
+    flexWrap: 'nowrap',
+    maxWidth: 740,
+    margin: '96px auto 126px auto',
+    minHeight: '100%',
+    maxHeight: 430,
+
+    ['@media(max-width: 580px)']: {
```
[the remark this answers is not in the document]
Comment (dhunanyan) [reply src/TestTask/style.ts:33]: Thanks for sharing, good to know. I've noticed that you use standard material-ui breakpoints. I tried to do the same, however in some instances decided to go for custom media queries for better look & feel.

File: src/TestTask/style.ts (hunk 3 above)
Comment (dhunanyan) [reply src/TestTask/style.ts:36]: Agree.

File: src/TestTask/style.ts (hunk 4 above)
Comment (dhunanyan) [reply src/TestTask/style.ts:7]: Thanks.

File: src/TestTask/components/Wallet/Wallet.tsx (hunk 5 above)
Comment (dhunanyan) [reply src/TestTask/components/Wallet/Wallet.tsx:88]: Thanks for flagging.

File: src/TestTask/components/Stats/Stats.stories.tsx (hunk 6 above)
Comment (dhunanyan) [reply src/TestTask/components/Stats/Stats.stories.tsx:13]: Noted.

File: src/TestTask/components/Header/Header.tsx (hunk 7 above)
Comment (dhunanyan) [reply src/TestTask/components/Header/Header.tsx:1]: I noticed that components from the mockup were similar to ones in the project, some of them were even identical. However having read the task instruction carefully, I realised, that I was expected to create components rather than use available ones. This was also the reason of creating a whole new directory for the task.

File: src/TestTask/App.stories.tsx (hunk 8 above)
Comment (dhunanyan) [reply src/TestTask/App.stories.tsx:19]: Noted.

File: src/TestTask/components/Amount/Amount.tsx (hunk 9 above)
Comment (dhunanyan) [reply src/TestTask/components/Amount/Amount.tsx:1]: Noted. Please see my response to this comment: https://github.com/invariant-labs/webapp/pull/297#discussion_r882119874

File: src/TestTask/components/AnimatedButton/AnimatedButton.tsx (hunk 10 above)
Comment (dhunanyan) [reply src/TestTask/components/AnimatedButton/AnimatedButton.tsx:1]: Noted. Please see my response to this comment: https://github.com/invariant-labs/webapp/pull/297#discussion_r882119874

File: src/TestTask/components/Plot/Plot.stories.tsx (hunk 11 above)
Comment (dhunanyan) [reply src/TestTask/components/Plot/Plot.stories.tsx:10]: Understand, thank you

File: src/TestTask/components/CalculatorSwitch/CalculatorSwitch.tsx (hunk 12 above)
Comment (dhunanyan) [reply src/TestTask/components/CalculatorSwitch/CalculatorSwitch.tsx:28]: Thanks.

File: src/TestTask/components/CalculatorValue/CalculatorValue.tsx (hunk 13 above)
Comment (dhunanyan) [reply src/TestTask/components/CalculatorValue/CalculatorValue.tsx:32]: The intention was to make the components more interactive, however it looks like it was unnecessary and could make no sense for those who are aware about bitcoins.

File: src/TestTask/components/Info/style.ts (hunk 14 above)
Comment (dhunanyan) [reply src/TestTask/components/Info/style.ts:28]: Thanks, noted.

File: src/TestTask/components/Info/style.ts (hunk 15 above)
Comment (dhunanyan) [reply src/TestTask/components/Info/style.ts:30]: Understand.

File: src/TestTask/components/Wallet/Wallet.tsx (hunk 17 above)
Comment (dhunanyan) [reply src/TestTask/components/Wallet/Wallet.tsx:72]: Agree.

**Comment by dhunanyan** [comment]
Thanks for your time and detailed comments, they were very useful.
